### ISSUE TITLE
Add support for `removed` block

### ISSUE THREAD

**Issue by nevelis** [hashicorp/terraform#34402]
### Terraform Version

```shell
Terraform v1.6.5
on linux_amd64
+ provider registry.terraform.io/hashicorp/aws v5.28.0
+ provider registry.terraform.io/hashicorp/external v2.3.1
+ provider registry.terraform.io/hashicorp/null v3.2.1
```

### Use Cases

The proposal is to add a block to mark state to be removed from the state file.

The last few major Terraform releases has been good to us.  We had `moved`, we got `import`, we got for_each over import blocks.  With these, and the `-replace` flag, we can now prepare plans that encapsulate the entire transition of the infrastructure state, instead of doing:

* Use `terraform apply -replace=x` over `terraform taint x`
* Use `import { from = x, to = y }` over `terraform import x y`
* Use `moved { from = x, to = y }` over `terraform state mv x y`

... but we have nothing for `terraform state rm x`

We support moving objects around, we support importing, but we cannot support un-importing.  Being able to detach an object from being managed by Terraform during the application of a plan would prevent the last of our remaining needs to manually manipulate the state file.  For now. :) 

### Attempted Solutions

Initial condition:
```
# main.tf:

resource "aws_vpc" "myvpc" {
   ...
}

resource "aws_vpc" "theirvpc" {
   ...
}

import {
  id = "xxx"
  to = aws_vpc.othervpc
}

moved {
  from = aws_vpc.othervpc
  to   = aws_vpc.theirvpc
}
```

Now lets say that we have released this, and we either decide we've imported the wrong `othervpc` that has been moved to `theirvpc`.  If we remove `aws_vpc.their_vpc` from `main.tf`, the object will be removed.  In order to prevent this, we would do `terraform state rm aws_vpc.theirvpc` before applying these changes.

### Proposal

```
# main.tf:

resource "aws_vpc" "myvpc" {
   ...
}

removed {
  from = aws_vpc.theirvpc
}

```

```
terraform apply
```

**Comment by jbardin** [comment]
Hi @nevelis,

A `removed` block feature will be available in the v1.7. You can see the first beta release notes here: https://github.com/hashicorp/terraform/releases/tag/v1.7.0-beta1

Thanks!

**Comment by nevelis** [comment]
YOU ARE AMAZING.  Love your work! :grin:
